c10/core/TensorImpl.cpp

```diff
@@ -122,12 +122,34 @@ TensorImpl::TensorImpl(
   }
 }
 
+// This is currently used to construct view tensors for nested_tensor impls
 TensorImpl::TensorImpl(
+    ImplType type,
     DispatchKeySet key_set,
     const caffe2::TypeMeta data_type,
     c10::optional<c10::Device> device_opt)
     // NOLINTNEXTLINE(performance-move-const-arg)
-    : TensorImpl({}, key_set, data_type, std::move(device_opt)) {}
+    : storage_(),
+      pyobj_interpreter_(nullptr),
+      pyobj_(nullptr),
+      storage_offset_(0),
+      numel_(0),
+      data_type_(data_type),
+      device_opt_(device_opt),
+      key_set_(key_set - c10::python_ks) { // See [Note: Python key removal]{
+  init_bitfields();
+  // Inference tensor doesn't have version counter.
+  if (!is_inference()) {
+    version_counter_ = VariableVersion(/*version=*/0);
+  }
+}
+
+TensorImpl::TensorImpl(
+    DispatchKeySet key_set,
+    const caffe2::TypeMeta data_type,
+    c10::optional<c10::Device> device_opt)
+    // NOLINTNEXTLINE(performance-move-const-arg)
+    : TensorImpl(Storage{}, key_set, data_type, std::move(device_opt)) {}
```

Comment:
The change to this constructor was calling
going from `TensorImpl({}, key_set, data_type, std::move(device_opt)) {}` -> `TensorImpl(Storage{}, key_set, data_type, std::move(device_opt)) {}`

There was ambiguity on the dtype of the first member and this was call the correct constructor.